grails-app/services/org/pih/warehouse/dashboard/NumberDataService.groovy

```diff
@@ -210,6 +211,28 @@ class NumberDataService {
         return new NumberData(openStockRequests[0], "${urlContextPath}/stockMovement/list?direction=OUTBOUND&sourceType=ELECTRONIC")
     }
 
+    @Cacheable(value = "dashboardCache", key = { "getRequestsPendingApproval-${location?.id}-${currentUser?.id}" })
+    NumberData getRequestsPendingApproval(Location location, User currentUser) {
+        def openStockRequests = Requisition.executeQuery("""
+            SELECT COUNT(distinct r.id) FROM Requisition r
+            WHERE r.origin = :location
+            AND r.sourceType = :sourceType
+            AND r.status  = :status
+            """,
+                [
+                        'location': location,
+                        'sourceType' : RequisitionSourceType.ELECTRONIC,
+                        'status' : RequisitionStatus.PENDING_APPROVAL,
+                ])
```

Comment:
Wouldn't it be better to use createCritera here?

Reply:
I would go even further and give the data service method a shot (as I can see we already have the `RequisitionDataService`, so this would require implementing only a method)
```groovy
int countByOriginAndSourceTypeAndStatus(Location origin, RequisitionSourceType sourceType, RequisitionStatus status)
```

Reply:
Previously I was just following the example of other NumberService methods which compute data with pure SQL statements, but you are right something like a `createCriteria()` looks a bit more friendly so I went with that approach

Reply:
I also tested your suggestion @kchelstowski with `countByOriginAndSourceTypeAndStatus()` but I realized that we need to cache this indicator and the keys are `location.id` and `currentUser.id` so either way i would need to call this `RequisitionDataService` method in a `NumberDataService` so it creates a little more clutter and methods that we have to maintain so I decided to go with `createCriteria()`

Reply:
Although now that I think about it, I remember you mentioning something that GORM Data Service methods are much faster than the standard criteria ones, so this might be a reason to use the DataService instead?

Reply:
@drodzewicz could you elaborate a bit on the fact that you couldn't use the data service method here? I don't mind the `createCriteria()` approach, but wanted to understand why the data service method didn't fit there?

Reply:
> Although now that I think about it, I remember you mentioning something that GORM Data Service methods are much faster than the standard criteria ones, so this might be a reason to use the DataService instead?

I think I've never said they are faster - it should probably evaluate to the same SQL query as the data service method, but imho they are just more readable and they don't require that much of "boilerplate" code - e.g. the `createCriteria` approach here took more lines than one-liner data service method.